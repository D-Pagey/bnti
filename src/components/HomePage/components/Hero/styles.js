import styled from 'styled-components';
import { margin, padding } from 'polished';
import { border, colour, fontSize, spacing, mediaQuery } from '../../../../tokens';

export const Wrapper = styled.div`
    align-items: center;
    display: flex;
    flex-direction: column;
    justify-content: center;
    ${margin(0, 'auto')}
    max-width: 375px;
    ${padding(spacing.large, 0)};

    @media ${mediaQuery.min.medium} {
        flex-direction: row;
        max-width: 100%;
    }
`;

export const TitleWrapper = styled.div`
    @media ${mediaQuery.min.medium} {
        align-items: center;
        flex-direction: column;
        display: flex;
    }
`;

export const Subtitle = styled.h3`
    color: ${colour.blue};
    font-size: ${fontSize.huge};
    ${margin(0, 0, '2rem', 0)};
    max-width: 375px;
    text-align: center;
`;

export const Button = styled.button.attrs({
    type: 'button'
})`
    background: linear-gradient(180deg, #20fff2 0%, rgba(255, 255, 255, 0) 100%), #00a7cb;
    border: 0;
    border-radius: ${border.radiusLarge};
    cursor: pointer;
    font-size: ${fontSize.medium};
    ${margin('0', '0', spacing.medium)};
    opacity: 0.9;
    ${padding('1rem', '1.35rem')};
`;

export const Games = styled.div`
    align-items: center;
    display: flex;
    flex-wrap: wrap;
    justify-content: center;

    @media ${mediaQuery.min.medium} {
        width: 450px;
    }
`;

export const GameLogo = styled.img`
    height: auto;
    ${margin('10px', '20px')};
    width: ${({ width }) => width};
`;
