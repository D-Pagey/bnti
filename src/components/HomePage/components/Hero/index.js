import React from 'react';
import { NavLink } from 'react-router-dom';
import Media from 'react-media';
import { mediaQuery } from '../../../../tokens';
import logoApex from './assets/logo-apex.png';
import logoFortnite from './assets/logo-fortnite.png';
import logoPubg from './assets/logo-pubg.png';
import * as S from './styles';

const Hero = () => (
    <S.Wrapper>
        <S.TitleWrapper>
            <S.Subtitle>Discover a new way to play your favourite games</S.Subtitle>

            <Media
                query={mediaQuery.min.medium}
                render={() => (
                    <NavLink to="/tournaments">
                        <S.Button>Find tournaments</S.Button>
                    </NavLink>
                )}
            />
        </S.TitleWrapper>

        <S.Games>
            <S.GameLogo src={logoPubg} alt="pubg" width="250px" />
            <S.GameLogo src={logoApex} alt="apex" width="150px" />
            <S.GameLogo src={logoFortnite} alt="fortnite" width="150px" />
        </S.Games>

        <Media
            query={mediaQuery.max.medium}
            render={() => (
                <NavLink to="/tournaments">
                    <S.Button>Find tournaments</S.Button>
                </NavLink>
            )}
        />
    </S.Wrapper>
);

export default Hero;
