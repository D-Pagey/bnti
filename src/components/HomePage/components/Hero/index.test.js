import React from 'react';
import { render } from 'react-testing-library';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import Hero from '.';

const renderWithRouter = (ui) => render(<Router history={createMemoryHistory()}>{ui}</Router>);

describe('Hero component', () => {
    it('should render', () => {
        const { container } = renderWithRouter(<Hero />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it.each`
        viewportWidth | testId
        ${500}        | ${'tournamentsLinkMaxMedium'}
        ${800}        | ${'tournamentsLinkMinMedium'}
    `('should render with navbar', ({ viewportWidth, testId }) => {
        setMatchMedia(viewportWidth);
        const { getByTestId } = renderWithRouter(<Hero />);
        getByTestId(testId);
    });
});
