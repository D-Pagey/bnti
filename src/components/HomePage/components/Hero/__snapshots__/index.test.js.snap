// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero component should render 1`] = `
.c0 {
  background-color: #353535;
  padding: 1rem 0;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 10px auto;
  width: 950px;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 450px;
}

.c3 {
  color: #00A7CB;
  font-size: 36px;
  margin: 0 0 2rem 0;
  text-align: center;
}

.c4 {
  background: linear-gradient(180deg,#20fff2 0%,rgba(255,255,255,0) 100%),#00a7cb;
  border: 0;
  border-radius: 33px;
  cursor: pointer;
  font-size: 1.25rem;
  opacity: 0.9;
  padding: 1rem 1.35rem;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 430px;
}

.c6 {
  height: auto;
  margin: 10px 20px;
  width: 250px;
}

.c7 {
  height: auto;
  margin: 10px 20px;
  width: 150px;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <h3
        class="c3"
      >
        Discover a new way to play your favourite games
      </h3>
      <a
        href="/tournaments"
      >
        <button
          class="c4"
          type="button"
        >
          Find games
        </button>
      </a>
    </div>
    <div
      class="c5"
    >
      <img
        alt="pubg"
        class="c6"
        src="logo-pubg.png"
        width="250px"
      />
      <img
        alt="apex"
        class="c7"
        src="logo-apex.png"
        width="150px"
      />
      <img
        alt="fortnite"
        class="c7"
        src="logo-fortnite.png"
        width="150px"
      />
    </div>
  </div>
</div>
`;
